### PR TITLE
Hierarchies-react: Auto expand node when filter is applied

### PR DESCRIPTION
**Pull request by jasdom**
Changes needed for https://github.com/iTwin/viewer-components-react/issues/923.

### CODE REVIEW

File: packages/hierarchies-react/src/presentation-hierarchies-react/internal/TreeModel.ts [hunk 1]
```diff
@@ -151,6 +151,9 @@ export namespace TreeModel {
 
   export function setInstanceFilter(model: TreeModel, nodeId: string | undefined, filter?: GenericInstanceFilter): boolean {
     return updateForReload(model, nodeId, (node) => {
+      if (isTreeModelHierarchyNode(node)) {
```
Comment (saskliutas) [inline packages/hierarchies-react/src/presentation-hierarchies-react/internal/TreeModel.ts:154]: Do we want to expand collapsed nodes when filter is removed?

Comment (grigasp) [reply packages/hierarchies-react/src/presentation-hierarchies-react/internal/TreeModel.ts:154]: no :)

Comment (jasdom) [reply packages/hierarchies-react/src/presentation-hierarchies-react/internal/TreeModel.ts]: Fixed.

File: .changeset/thirty-vans-knock.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@itwin/presentation-hierarchies-react": minor
```
Comment (saskliutas) [inline .changeset/thirty-vans-knock.md:2]:
```suggestion
"@itwin/presentation-hierarchies-react": patch
```

**Review by saskliutas — APPROVED** [review]

**Pull request merged**